app/types/index.ts

```diff
@@ -0,0 +1,23 @@
+import {RequestObj} from './../request_list'
+export type POPUP_PROPS = {
+  tabId : number,
+  tabUrl : string,
+  enabled : boolean,
+  requests : Array<RequestObj>,
+  errorMessage : string
+}
+
+export interface PopUpState {
+  enabled: boolean;
+  errorMessage ?: string;
+  requests: object;
+  tabUrl ?: string
+}
+
+export interface Action extends PopUpState{
```

Comment:
Why does action need to extend popupstate?

Reply:
since Action includes all the properties of popupstate and an additional `type` property, extended it.. Any suggestions?

Reply:
No, was just wondering. In general, if this is a redux action, it need not have to be a global match all type.